src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -1,47 +1,151 @@
-        ancestorTableVariable.makeFieldVariable("descendant_concept_id");
+    List<FieldVariable> groupBy = List.of(nameFieldVariable, idFieldVariable);
 
-    BinaryFilterVariable whereClause =
+    List<OrderByVariable> orderBy =
+        List.of(new OrderByVariable(nameFieldVariable, OrderByDirection.ASCENDING));
+
+    // ancestorTable is primary table for the subquery
+    var ancestorTablePointer = TablePointer.fromTableName(CONCEPT_ANCESTOR, tableNameGenerator);
+    var ancestorTableVariable = TableVariable.forPrimary(ancestorTablePointer);
+    var descendantFieldVariable = ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    // WHERE c.ancestor_concept_id = conceptId
+    BinaryFilterVariable ancestorClause =
         new BinaryFilterVariable(
-            ancestorTableVariable.makeFieldVariable("ancestor_concept_id"),
+            ancestorTableVariable.makeFieldVariable(ANCESTOR_CONCEPT_ID),
             BinaryFilterVariable.BinaryOperator.EQUALS,
             new Literal(conceptId));
+
+    // WHERE d.descendant_concept_id != conceptId
+    BinaryFilterVariable notSelfClause =
+        new BinaryFilterVariable(
+            ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID),
+            BinaryFilterVariable.BinaryOperator.NOT_EQUALS,
+            new Literal(conceptId));
+
+    // WHERE c.ancestor_concept_id = conceptId AND d.descendant_concept_id != conceptId
+    List<FilterVariable> clauses = List.of(ancestorClause, notSelfClause);
+    BooleanAndOrFilterVariable subqueryWhereClause =
+        new BooleanAndOrFilterVariable(BooleanAndOrFilterVariable.LogicalOperator.AND, clauses);
+
+    // (SELECT c.descendant_concept_id FROM concept_ancestor AS c
+    // WHERE c.ancestor_concept_id = conceptId)
     Query subQuery =
-        new Query(List.of(descendantFieldVariable), List.of(ancestorTableVariable), whereClause);
-    SubQueryFilterVariable subQueryFilterVariable =
-        new SubQueryFilterVariable(idFieldVariable, SubQueryFilterVariable.Operator.IN, subQuery);
-    Query query =
         new Query(
-            List.of(nameFieldVariable, idFieldVariable),
-            List.of(conceptTableVariable),
-            subQueryFilterVariable);
+            List.of(descendantFieldVariable), List.of(ancestorTableVariable), subqueryWhereClause);
+
+    // WHERE c.concept_id IN subQuery
+    SubQueryFilterVariable where =
+        new SubQueryFilterVariable(idFieldVariable, SubQueryFilterVariable.Operator.IN, subQuery);
+
+    Query query = new Query(select, tables, where, groupBy, orderBy);
+    return query.renderSQL(platform);
+  }
+
+  /**
+   * Generate a query that retrieves the domain id of the concept specified by the given conceptId.
+   *
+   * <p>SELECT c.domain_id FROM concept AS c WHERE c.concept_id = conceptId
+   */
+  public static String retrieveDomainId(
+      Integer conceptId, TableNameGenerator tableNameGenerator, CloudPlatformWrapper platform) {
```

Comment:
Since `conceptId` can't be `null`, it should use the unboxed type `int`
```suggestion
      int conceptId, TableNameGenerator tableNameGenerator, CloudPlatformWrapper platform) {
```